src/display/canvas.js

```diff
@@ -325,7 +325,6 @@ function compileType3Glyph(imgData) {
 
       coords.push(p % width1);
       coords.push((p / width1) | 0);
-      --count;
```

Comment:
removing this is not preferable: could you try to move `--count` after `points[p]` is set to 0. you can see that at line 323 it may or may not do that, so in the else branch let's add the check `if (!points[p]) --count;`

Reply:
Sure, let's see how this works out on the bots :-)

(If it turns out to be more complicated, someone who actually understands this code will have to take over fixing the bug though.)

Reply:
It seems that this worked out just fine; does this mean that the patch is good to go now?